### DISCUSSION TITLE
Module Federation / Nx.dev / Remix.run

### DISCUSSION THREAD

**Discussion by ben-zabloski** [module-federation/module-federation-examples#2333] [General]
Watched the [Nx / Module Federation live stream](https://youtu.be/wGc-vx7QZ6Q) today and was curious about... combining everything 😅 

I would love to take a crack at:
- Having `@nrwl/remix:host` generators in the same way as `nx g @nrwl/react:host host --remotes=shop,cart,about` works - https://nx.dev/recipe/faster-builds#faster-builds-with-module-federation
- Seeing Remix augmented with Native Federation:
  - https://twitter.com/ScriptedAlchemy/status/1562585403240042496 (@ScriptedAlchemy - Not sure if this was using Native Federation or the Webpack version? 🤔).
    - I've seen this example from @jacob-ebey: https://github.com/jacob-ebey/esbuild-federation-example
  - Native Federation:
    - https://twitter.com/ScriptedAlchemy/status/1564015417651351553
    - https://github.com/module-federation/module-federation-examples/tree/master/native-federation-react

Have you heard if Remix is looking at supporting module federation in any official capacity or would we have to apply patches to make things happen? 🤔

Just getting excited about using everything here in a large-scale project 😁

**Comment by ScriptedAlchemy** [comment]
Remix has a bundler agnostic version in the works, so you can just use webpack not have to use esbuild. 
if you want to use esbuild then you'll need native-federation. 

Jacob co-authored half the MF ecosystem with me, so there will likely be catered support for the tech - unlike next.js which actively suppresses it.
